vendor/ocaml-4.13-extended/parsetree.mli

```diff
@@ -993,3 +993,9 @@ and directive_argument_desc =
   | Pdir_int of string * char option
   | Pdir_ident of Longident.t
   | Pdir_bool of bool
+
+type repl_phrase =
+  {
+    prepl_phrase : toplevel_phrase;
+    prepl_output : string;
+  }
```

Comment:
Is this record useful at all ? It's causing problems if a block starts with something that's not a phrase.
I think it could be a `` [ `Command of string | `Output of string ] list ``.

Reply:
I'm not sure we should allow this kind of blocks (maybe we should idk), I have the feeling `mdx` is keeping this representation because it doesn't need something more precise and doesn't need to maintain an AST like ocamlformat.

Reply:
This is specifically causing problems with repl-files starting with an empty line. Perhaps we can at least skip empty lines ?

Reply:
@NathanReb I didn't find anything about preserving empty lines and such in the spec, what is your take on this?

Reply:
I'm unsure what MDX allows here but I think we don't take a huge risk by doing by accepting whitespaces before the first phrase in toplevel blocks in the input but stripping them out in the output.

If there is anything else than this before the first phrase I would consider the block invalid and simply not format it.

Does that sound like a reasonable behaviour to you?

Note that I would also be perfectly happy if the blocks were required to start with the first phrase on the first line without any extra whitespace!